File: linear_regression.R
# Import the dataset
dataset = read.csv('Dealership_Data.csv')

# Split the dataset into the Training set and Test set
library(caTools)
set.seed(123)
split = sample.split(dataset$Sell.Price, SplitRatio = 3/4)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Fit Simple Linear Regression to the Training set
regressor = lm(formula = Sell.Price ~ List.Price,
               data = training_set)

# Predict the Test set results
y_pred = predict(regressor, newdata = test_set)

# Visualize the Training set results
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$List.Price, y = training_set$Sell.Price),
             colour = 'red') +
  geom_line(aes(x = training_set$List.Price, y = predict(regressor, newdata = training_set)),
            colour = 'blue') +
  ggtitle('List Price vs Sale Price (Training set)') +
  xlab('List Price') +
  ylab('Sell Price')

# Visualize the Test set results
ggplot() +
  geom_point(aes(x = test_set$List.Price, y = test_set$Sell.Price),
             colour = 'red') +
  geom_line(aes(x = training_set$List.Price, y = predict(regressor, newdata = training_set)),
            colour = 'blue') +
  ggtitle('List Price vs Sale Price (Test set)') +
  xlab('List Price') +
  ylab('Sell Price')

# Print slope and intercept of the model
print(regressor$coefficients)

# Print a prediction
print(predict(regressor, newdata = data.frame('List.Price' = c(20))))

File: linear_regression/linear_regression.py
"""
Created on Fri Feb  7 19:14:31 2025

@author: allyschmidt
"""

# Import the libraries
import matplotlib.pyplot as plt
import pandas as pd

# Import the dataset
dataset = pd.read_csv('Dealership_Data.csv')
X = dataset.iloc[:, :-1].to_numpy()
y = dataset.iloc[:, -1].to_numpy()

# Split the dataset into the Training set and Test set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/4,
                                                    random_state = 0)

# Train the Simple Linear Regression model on the Training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Predict the Test set results
y_pred = regressor.predict(X_test)

# Visualize the Training set results
plt.scatter(X_train, y_train, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('List Price vs Sale Price (Training set)')
plt.xlabel('List Price')
plt.ylabel('Sale Price')
plt.show()

# Visualize the Test set results
plt.scatter(X_test, y_test, color='red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('List Price vs Sale Price (Test set)')
plt.xlabel('List Price')
plt.ylabel('Sale Price')
plt.show()

# Print slope and intercept of the model
print('Slope:',regressor.coef_)
print('Intercept:', regressor.intercept_)

# Print a prediction
print('Predicted sale price of an item listed at $20k:', regressor.predict([[20]]))

